packages/babel-types/src/definitions/core.ts

```diff
@@ -1191,6 +1206,9 @@ defineType("AssignmentPattern", {
         "ObjectPattern",
         "ArrayPattern",
         "MemberExpression",
+        "TSAsExpression",
+        "TSTypeAssertion",
```

Comment:
Can AssignmengPattern.left be a TSTypeAssertion?

```ts
[<number>a = 1] = [42];
{<number>a = 1} = [42];
```
is invalid in TS nightly: https://www.typescriptlang.org/play?jsx=0&ts=4.7.0-dev.20220502#code/NoHgdgrgtgRgpgJwHwEMAEBeNBGAuptYAFgCZcBuAKAG9xp5l0tsBfA4s8oA